### ISSUE TITLE
[Feature] addon.tab.getScratchVM() (Function.prototype.bind trap)

### ISSUE THREAD

**Issue by WorldLanguages** [ScratchAddons/ScratchAddons#75]
**Is your feature request related to a problem? Please describe.**

Many addons that interact with the editor will likely want a direct reference to the Scratch VM object.

**Describe the solution you'd like**

A new method inside `addon.tab` that resolves to the VM object when it's found.

**Describe alternatives you've considered**

Make the promise reject if the VM won't ever be found? Not sure how that would work.

**Additional context**

Code here: https://gist.github.com/BoomerScratch/0592337050810244236d69a4267b3913

**Comment by BoomerScratch** [comment]
> Make the promise reject if the VM won't ever be found? Not sure how that would work.

Currently, the promise will never resolve if the VM is not found. Also, you need to inject the content script that gets the VM _before the page loads_. Can that be done?

**Comment by WorldLanguages** [comment]
Yep it can (don't forget Tampermonkey by itself is an extension, if TM can, we also can) but it might require another script to load on `document_start`. Currently the content script runs at `document_idle` and the injected modules run in `defer` mode.
